typing/typemod.ml

```diff
@@ -3281,14 +3297,20 @@ let report_error ~loc _env = function
         "@[Multiple definition of the %s name %s.@ \
          Names must be unique in a given structure or signature.@]"
         (Sig_component_kind.to_string kind) name
-  | Non_generalizable typ ->
+  | Non_generalizable { variable; expression } ->
+      prepare_for_printing [ variable; expression ];
       Location.errorf ~loc
         "@[The type of this expression,@ %a,@ \
-           contains type variables that cannot be generalized@]" type_scheme typ
-  | Non_generalizable_module mty ->
+         contains the non-generalizable type variable %a@]"
+        prepared_type_scheme expression
+        prepared_type_scheme variable
```

Comment:
Idea:  Should we point to the manual section on non-generalizable type variables?

Reply:
Do you mean 5.1.2/5.1.3 (I think we need to refer to both sections)?

We could say: "See manual sections 5.1.2 and 5.1.3"?

Reply:
Using the `6.1` section as you have done seems appropriate to me.